### DISCUSSION TITLE
[Need help] Termux + xfce4  occupy 3GB memory then release in 5 seconds, sometimes termux killed by android system

### DISCUSSION THREAD

**Discussion by diyism** [termux/termux-app#2678] [Q&A]
I've successfully intalled termux, alpine, xfce4, chromium into my phone,
https://github.com/diyism/TermuxXserver
most times it works very well, but 20% of times the Termux app will be killed by the android system.

reproduction procedure:
1. Install termux into my phone
2. bash TermuxAlpine.sh
3. startalpine
4. apk add x11vnc xfce4 xvfb chromium
5. start x11vnc and Xvfb:  while true; do nohup /usr/bin/x11vnc -noxfixes -usepw -repeat -loop -create -noshm -gone 'killall Xvfb' -env X11VNC_CREATE_STARTING_DISPLAY_NUMBER=0 -env X11VNC_CREATE_GEOM=2560x1688x16 >/dev/null 2>&1; done &
at this step it seems the xfce4 won't start.

6. from another phone, use androidVNC client app to connect my Termux phone,
at this step the xfce4 will be started by the x11vnc process,
and I can see the Termux phone's memory be occupied 3GB first, 
then after 5 seconds, the 3GB memory will be released totally.
but in the 5 seconds, there are 20% chance the android system will kill the Termux app.

7. If the Termux phone's android system didn't kill the termux app, then I can start the chromium browser in my androidVNC phone,
the chromium works well without any subsequent problem.

Anyone has a hint to avoid the android system to kill the Termux app?

**Comment by ghost** [comment]
This is likely a memory leak in Alpine when starting `xfce4-session`. you'd need to check what process is hogging the memory and causes Termux to crash. use `htop` or `top` to monitor processes and it's memory usage. once you start `xfce4-session`, quickly go back to `(h)top` in another session to see what's the highest memory usage compared to others. then you may submit what process is causing the leak

`xfce4-power-manager` is likely the reason for this leak AFAIK. i can tell by getting a dialog box about `xfce4-power-manager` applet crash then retries itself to spawn until memory hogs

**Reply by diyism** [reply]
Thanks, finally I found it's the process of  "xfdesktop" occuped 3GB memory, so now I get rid of xfce and to use "nwrkbiz/android-xserver" to replace "x11vnc and xorg-server-xvfb", the xServer apk create a display at ":0", and I can run and see any linux application after "export DISPLAY=:0"
